File: rust/src/resources/api_keys.rs
use std::sync::Arc;

use reqwest::Method;

use crate::error::Result;
use crate::models::common::{ApiKey, ApiKeyCreateRequest, ApiKeyCreateResponse};
use crate::transport::Transport;

pub struct ApiKeysResource {
    transport: Arc<Transport>,
}

impl ApiKeysResource {
    pub(crate) fn new(transport: Arc<Transport>) -> Self {
        Self { transport }
    }

    pub async fn create(&self, request: &ApiKeyCreateRequest) -> Result<ApiKeyCreateResponse> {
        self.transport
            .request(Method::POST, "/v3/api-keys", Some(request))
            .await
    }

    pub async fn list(&self) -> Result<Vec<ApiKey>> {
        self.transport
            .request(Method::GET, "/v3/api-keys", None::<&()>)
            .await
    }

    pub async fn get(&self, id: &str) -> Result<ApiKey> {
        self.transport
            .request(
                Method::GET,
                &format!("/v3/api-keys/{}", id),
                None::<&()>,
            )
            .await
    }

    pub async fn revoke(&self, id: &str) -> Result<serde_json::Value> {
        self.transport
            .request(
                Method::POST,
                &format!("/v3/api-keys/{}/revoke", id),
                None::<&()>,
            )
            .await
    }

    pub async fn delete(&self, id: &str) -> Result<()> {
        self.transport
            .request::<serde_json::Value>(
                Method::DELETE,
                &format!("/v3/api-keys/{}", id),
                None::<&()>,
            )
            .await?;
        Ok(())
    }
}

File: go/api_keys.go
package aegis

import (
	"context"
	"fmt"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
	"github.com/ai-aegis-yatav/aegis-sdk/go/models"
)

type ApiKeysService struct {
	t *internal.Transport
}

func (s *ApiKeysService) Create(ctx context.Context, req models.ApiKeyCreateRequest) (*models.ApiKeyCreateResponse, error) {
	var resp models.ApiKeyCreateResponse
	if err := s.t.Do(ctx, "POST", "/v1/api-keys", req, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

func (s *ApiKeysService) List(ctx context.Context, params models.ListParams) (*models.PaginatedResponse[models.ApiKey], error) {
	var resp models.PaginatedResponse[models.ApiKey]
	path := buildPath("/v1/api-keys", params.ToQuery())
	if err := s.t.Do(ctx, "GET", path, nil, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

func (s *ApiKeysService) Get(ctx context.Context, id string) (*models.ApiKey, error) {
	var resp models.ApiKey
	if err := s.t.Do(ctx, "GET", fmt.Sprintf("/v1/api-keys/%s", id), nil, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

func (s *ApiKeysService) Revoke(ctx context.Context, id string) error {
	return s.t.Do(ctx, "POST", fmt.Sprintf("/v1/api-keys/%s/revoke", id), nil, nil)
}

func (s *ApiKeysService) Delete(ctx context.Context, id string) error {
	return s.t.Do(ctx, "DELETE", fmt.Sprintf("/v1/api-keys/%s", id), nil, nil)
}
